RPIPs/RPIP-60.md

```diff
@@ -1,52 +0,0 @@
----
-rpip: 60
-title: Protocol Upgrade Guardrails
-description: Introduces a delay after protocol upgrades have been confirmed but prior to them coming into effect.
-author: Valdorff (@Valdorff)
-discussions-to: TBD
-status: Draft
-type: Protocol
-category: Core
-created: 2024-03-08
-requires: 
-tags: tokenomics-2024, tokenomics-content
----
-
-## Abstract
-
-Regularly scheduled protocol upgrades now have a delay between when they are passed, and when they are executed. 
-
-The security council may trigger a faster upgrade with a shorter delay, but there are limitations on when they may use this power. This security council power has a maximum number of uses that may be set by the pDAO. The security council cannot use the fast upgrade power if they have no remaining uses, and each use removes one seal. Seals may be restored via pDAO vote. 
-
-Additionally the security council may veto contract upgrades. The veto power should not be used lightly, and misuse may result in replacement of the security council by the pDAO.
-
-## Specification
-
-### Protocol upgrades
-- All protocol upgrades SHALL have a `default_upgrade_delay` delay between when they are passed and when they are executed
-- The security council SHALL have a limited-use power to use a shorter delay of `fast_upgrade_delay`
-  - This power SHALL be usable if `fast_upgrade_seal_count` > 0 
-  - `fast_upgrade_seal_count` SHALL be decremented by one upon using this power
-  - The pDAO MAY include a change to `fast_upgrade_seal_count` in a voted upgrade*
-  - The pDAO MAY change `fast_upgrade_seal_count` via vote**
-  - The security council SHALL use this power if and only if:
```

Comment:
Can we add something here that smart contract upgrade code:
* Has been public for at least x time? 
* Is verified and such on etherscan?

Maybe these can be taken as obvious assumptions? Not sure.

Thinking about how this actually works, if with requires a pDAO vote, those are like 2 weeks lead time, so maybe a requirement that the upgrade code is _complete and public_ during that pDAO vote? Like, its less good to have pDAO voting on something that's not been written yet.

Reply:
I'm not too concerned about pDAO vote's relationship to code. Our spec is generally higher level than our code by quite a lot, so it shouldn't impact the vote.

If it has not followed pDAO process, or the code cannot otherwise be checked for security -- those should not get proposed by the oDAO. If they are proposed without any verification, I would consider that malicious action from the oDAO and grounds for a security council veto.